datumaro/plugins/coco_format/converter.py

```diff
@@ -451,6 +453,67 @@ def save_annotations(self, item):
 
             self.annotations.append(elem)
 
+class _StuffConverter(_InstancesConverter):
+    pass
+
+class _PanopticConverter(_TaskConverter):
+    def save_categories(self, dataset):
+        label_categories = dataset.categories().get(AnnotationType.label)
+        if label_categories is None:
+            return
+
+        for idx, cat in enumerate(label_categories.items):
+            self.categories.append({
+                'id': 1 + idx,
+                'name': cast(cat.name, str, ''),
+                'supercategory': cast(cat.parent, str, ''),
+            })
+
+    def save_annotations(self, item):
+        if not item.has_image:
+            return
+
+        ann_filename = item.id + CocoPath.PANOPTIC_EXT
+
+        def id2bgr(id_map):
+            id_map_copy = id_map.copy()
+            bgr_shape = tuple(list(id_map.shape) + [3])
+            bgr_map = np.zeros(bgr_shape, dtype=np.uint8)
+            for i in range(3):
+                bgr_map[..., 2-i] = id_map_copy % 256
+                id_map_copy //= 256
+            return bgr_map
+
+        segments_info = list()
+        masks = []
+        for ann in item.annotations:
+            if ann.type != AnnotationType.mask:
+                continue
+
+            segment_info = {}
+            segment_info['id'] = ann.id
+            segment_info['category_id'] = cast(ann.label, int, -1) + 1
+            segment_info['area'] = float(ann.get_area())
+            segment_info['bbox'] = [float(p) for p in ann.get_bbox()]
+            segment_info['iscrowd'] = cast(ann.attributes.get("is_crowd"), int, 0)
+            segments_info.append(segment_info)
+            masks.append(ann)
+
+        if masks:
+            pan_format = mask_tools.merge_masks(
+                ((m.image, m.id) for m in masks),
+                start=np.zeros(item.image.size, dtype=np.uint32))
+            save_image(osp.join(self._context._segmentation_dir, ann_filename),
+                id2bgr(pan_format), create_dir=True)
+
+        elem = {
+            'id': item.id,
```

Comment:
`id` does not exist in the official spec, what is the point in it?

Reply:
sorry, forgot to remove it. removed now